### DISCUSSION TITLE
Can Reactflow be used in SSG?

### DISCUSSION THREAD

**Discussion by NEWESTERS** [xyflow/xyflow#2924] [Q&A]
I am making personal blog about Javascript on [Astro](https://docs.astro.build/en/core-concepts/framework-components/#available-hydration-directives) framework and trying to use React flow to illustrate interactive graphs.

I'm trying to achieve static html generation of chart during build time and lazy hydration.

Here's code snippet for hydrate graph when it's visible with [client:visible](https://docs.astro.build/en/reference/directives-reference/#clientvisible) directive:
```tsx
import { RequestStateGraph } from "@articleAssets/state-machines";

<RequestStateGraph client:visible />
```

It renders perfectly with Javascript enabled:

<img width="400" alt="Снимок экрана 2023-03-18 в 14 17 41" src="https://user-images.githubusercontent.com/22481727/226102024-e12d77b6-2061-4047-8c98-7aca7741e7a3.png">

But when i disable Javascript it renders nothing:

<img width="400" alt="Снимок экрана 2023-03-18 в 14 18 45" src="https://user-images.githubusercontent.com/22481727/226102055-1a064758-faf8-4054-9995-8ff80128e67c.png">

ReactFlow wrappers renders fine during SSG, by my nodes are missing:

<img width="890" alt="Снимок экрана 2023-03-18 в 14 19 45" src="https://user-images.githubusercontent.com/22481727/226102100-1202c7c3-32d4-4bbe-b40e-89a86a8eb565.png">

The only approach, that i implemented now is to provide fallback image for graph and use it for SSR:

```tsx
const isSsr = typeof document === "undefined";

const MyGraph = ({ fallbackImageUrl, ...props }) => {
  if(isSsr) {
    return <img src={fallbackImageUrl} />
  } else {
    return <ReactFlow {...props} />
  }
}
```

Fallback image is generated manually and placed to static folder.

Can ReactFlow be used to render graph during SSG/SSR without hydration?

**Comment by ace-han** [comment]
I can confirm that with release 11.9.x it's working in *.astro page file 

https://github.com/wbkd/react-flow-example-apps/tree/main/reactflow-astro

It still fails to render while in *.mdx content page file...
